### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by RashidMenhas**
### Create a class Nameable.
- Implement a method called correct_name that will raise a NotImplementedError.

### Turn Person class to Nameable
- That Person class inherits from Nameable
-  That this class has a method correct_name implemented. It should simply return the name attribute.

### Prepare base Decorator
-  that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
Implement the correct_name method that returns the result of the correct_name method of the @nameable.

### CapitalizeDecorator:
- Create a class that inherits from the base Decorator class.
- Implement a method correct_name that capitalizes the output of @nameable.correct_name.

### TrimmerDecorator:
- Create a class that inherits from the base Decorator class.
- Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: capitalizedecorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require 'BaseDecorator'
```
Comment (Davidosky007) [inline capitalizedecorator.rb]: * Kindly import the right file properly as the `./basedecorator.rb` is in lowercase

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+require 'nameable'
+
+class Person < Nameable
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    super
```
Comment (Davidosky007) [inline person.rb]: * Kindly add `super( )` as I get an argument error as `nameable is not expecting any argument [Kindly read more on the super keyword](https://www.rubyguides.com/2018/09/ruby-super-keyword/) 
image

File: basedecorator.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super
```
Comment (Davidosky007) [inline basedecorator.rb]: * Kindly add `super( )` as I get an argument error as `nameable is not expecting any argument [Kindly read more on the super keyword](https://www.rubyguides.com/2018/09/ruby-super-keyword/) 
image

Comment (Davidosky007) [reply basedecorator.rb]: * This is what i get after fixing the issue 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @RashidMenhas 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Good looking readme file 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: capitalizedecorator.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require './basedecorator'
+
+class CapitalizeDecorator < BaseDecorator
+  def initialize(nameable)
+    super()
```
Comment (codepantha) [inline capitalizedecorator.rb]: - [ ] After implementing the changes requested by the previous reviewer, I'm sorry to say that your app is still not working as expected as you can see below 👇 
image


Please remove the `super()` line of code from your `basedecorator.rb`, `capitalizedecorator.rb` and `trimmerdecorator.rb` files and your app will run just fine 😎  like this 👇  
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @RashidMenhas 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BertrandConxy — APPROVED** [review]
Hi  @RashidMenhas, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- project works fine
- good PR description
- correct flow is used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by RashidMenhas** [comment]
brother please help me i face a problem on day 4 and i spent 2 days on it but didn't fix it please help me I will be very grateful to you it's about  (execution permission).Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: David Bassey ***@***.***> Date: 1/19/23  10:21 AM  (GMT+05:00) To: RashidMenhas/oop-school-library ***@***.***> Cc: Rashid Menhas ***@***.***>, Mention ***@***.***> Subject: Re: [RashidMenhas/oop-school-library] OOP school library: decorate a class (PR #2) 

@Davidosky007 requested changes on this pull request.



Hi @RashidMenhas 👋,

Good job so far!

There are some issues that you still need to work on to go to the next project but you are almost there!

Highlights



No linter errors 👍

Good looking readme file 👍



Required Changes ♻️

Check the comments under the review.

Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.



As described in the Code reviews limits policy you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this form.



In capitalizedecorator.rb:

> @@ -0,0 +1,12 @@

+require 'BaseDecorator'





Kindly import the right file properly as the ./basedecorator.rb is in lowercase









In person.rb:

> @@ -0,0 +1,30 @@

+require 'nameable'

+

+class Person < Nameable

+  attr_reader :id

+  attr_accessor :name, :age

+

+  def initialize(age, name = 'unknown', parent_permission: true)

+    super





Kindly add super( ) as I get an argument error as `nameable is not expecting any argument Kindly read more on the super keyword











In basedecorator.rb:

> @@ -0,0 +1,12 @@

+require 'nameable'

+

+class BaseDecorator < Nameable

+  def initialize(nameable)

+    super





Kindly add super( ) as I get an argument error as `nameable is not expecting any argument Kindly read more on the super keyword











In basedecorator.rb:

> @@ -0,0 +1,12 @@

+require 'nameable'

+

+class BaseDecorator < Nameable

+  def initialize(nameable)

+    super





This is what i get after fixing the issue







—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

[

{

***@***.***": "http://schema.org",

***@***.***": "EmailMessage",

"potentialAction": {

***@***.***": "ViewAction",

"target": "https://github.com/RashidMenhas/oop-school-library/pull/2#pullrequestreview-1261126907",

"url": "https://github.com/RashidMenhas/oop-school-library/pull/2#pullrequestreview-1261126907",

"name": "View Pull Request"

},

"description": "View this Pull Request on GitHub",

"publisher": {

***@***.***": "Organization",

"name": "GitHub",

"url": "https://github.com"

}

}

]

**Pull request merged**